lib/connection.js

```diff
@@ -158,7 +166,15 @@ class Connection extends EventEmitter {
     }
 
     const request = this.requestBuffer.shift();
-    const responder = request && request.pop();
+
+    let responder;
+    const req_id = response?.$req_id;
+    if (req_id) {
```

Comment:
This path should have unit test coverage

Reply:
Added test case when `$req_id` is defined and responses are received in reverse order